lib/amazonka-core/src/Amazonka/Sign/V4/Base.hs

```diff
@@ -308,4 +308,5 @@ normaliseHeaders =
     . Map.toAscList
     . Map.delete "authorization"
     . Map.delete "content-length"
+    . Map.delete "expect"
```

Comment:
Let's remove all of the headers blacklisted by botocore, so please remove the `user-agent` and `x-amzn-trace-id` headers too.

There are so many places where we have to do things just because an SDK does them that way, or someone once spoke to support, or a service doesn't quite behave to spec ([example](https://github.com/brendanhay/amazonka/blob/9c3594073a7787e27f13fa14c936f0fa5cb9f0e8/lib/amazonka-core/src/Amazonka/Sign/V4/Chunked.hs#L36-L42)). Can you please add a comment that we're doing this for R2's sake, and also link to https://github.com/boto/botocore/blob/d5b2e4ab4bc4ad84f8e0e568e70ddc8ab7f094a8/botocore/auth.py#L61-L65 ?

Reply:
Yeah I was lying in bed last night thinking I needed to have a much better comment in the code for this "simple" commit. :P Coming right up!

Reply:
Done